### PR TITLE
Rewrite `ckeditor5-style` to TypeScript

### LINKED ISSUE #13032: [TS] Rewrite ckeditor5-style into TypeScript
Part of #11704.

### PR DESCRIPTION
**Pull request by filipsobol**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Other (style): Rewrite `ckeditor5-style` to TypeScript. Closes #13032.

**🚨 IMPORTANT: Before merging this PR, change the target branch to `master` 🚨**

### CODE REVIEW

File: packages/ckeditor5-clipboard/package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "@ckeditor/ckeditor5-engine": "^36.0.1",
     "@ckeditor/ckeditor5-utils": "^36.0.1",
     "@ckeditor/ckeditor5-widget": "^36.0.1",
-    "lodash-es": "^4.17.11"
+    "lodash-es": "^4.17.15"
```
Comment (filipsobol) [inline packages/ckeditor5-clipboard/package.json:19]: This is the only package using `.11` while all others use `.15`

File: packages/ckeditor5-html-support/src/generalhtmlsupport.ts [hunk 2]
```diff
@@ -102,7 +102,7 @@ export default class GeneralHtmlSupport extends Plugin {
 	 * @param className The css class to add.
 	 * @param selectable The selection or element to update.
 	 */
-	private addModelHtmlClass( viewElementName: string, className: ArrayOrItem<string>, selectable: LimitedSelectable ) {
+	public addModelHtmlClass( viewElementName: string, className: ArrayOrItem<string>, selectable: LimitedSelectable ): void {
```
Comment (arkflpc) [inline packages/ckeditor5-html-support/src/generalhtmlsupport.ts:105]: Make it internal. I know it's used outside this package, but it shouldn't be used by anyone (`ckeditor5-style` is an exception).

File: packages/ckeditor5-html-support/src/generalhtmlsupport.ts [hunk 3]
```diff
@@ -124,7 +124,7 @@ export default class GeneralHtmlSupport extends Plugin {
 	 * @param className The css class to remove.
 	 * @param selectable The selection or element to update.
 	 */
-	private removeModelHtmlClass( viewElementName: string, className: ArrayOrItem<string>, selectable: LimitedSelectable ) {
+	public removeModelHtmlClass( viewElementName: string, className: ArrayOrItem<string>, selectable: LimitedSelectable ): void {
```
Comment (arkflpc) [inline packages/ckeditor5-html-support/src/generalhtmlsupport.ts:127]: Make it internal. I know it's used outside this package, but it shouldn't be used by anyone (`ckeditor5-style` is an exception).

File: packages/ckeditor5-style/src/style.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import StyleEditing from './styleediting';
+
+/**
+ * The style plugin.
+ *
+ * This is a "glue" plugin that loads the {@link module:style/styleediting~StyleEditing style editing feature}
+ * and {@link module:style/styleui~StyleUI style UI feature}.
+ *
+ * @extends module:core/plugin~Plugin
+ */
```
Comment (arkflpc) [inline packages/ckeditor5-style/src/style.ts:22]:
```suggestion
 */
```

File: packages/ckeditor5-style/src/styleconfig.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+   *
+   *   // ...
+   * ]
+   * ```
+   *
+   * **Note**: Configuring style definitions will automatically configure the
+   * {@glink features/general-html-support General HTML Support feature}. **You do not need to repeat the configuration in
+   * {@link module:html-support/generalhtmlsupport~GeneralHtmlSupportConfig}**.
+   */
+  definitions: Array<StyleDefinition>;
```
Comment (arkflpc) [inline packages/ckeditor5-style/src/styleconfig.ts:70]:
```suggestion
  definitions?: Array<StyleDefinition>;
```

File: packages/ckeditor5-style/src/styleconfig.ts [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+ * ```ts
+ * // This style will create <h2 class="foo">...</h2> in the document data.
+ * {
+ *   name: 'Example style',
+ *   element: 'h2',
+ *   classes: [ 'foo' ]
+ * }
+ * ```
+ */
+export type StyleDefinition = {
```
Comment (arkflpc) [inline packages/ckeditor5-style/src/styleconfig.ts:87]:
```suggestion
export interface StyleDefinition {
```

**Review by arkflpc — CHANGES_REQUESTED** [review]

**Review by arkflpc — APPROVED** [review]

**Pull request merged**